---
title: "Resubmission Graphs"
author: "RAZ"
date: "January 8, 2018"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
library("ggplot2")
library("DESeq2")
library("plyr")
library("dplyr")
library("tidyr")
library("tibble")
library("VennDiagram")
library("seqinr")
library("gridExtra")
library("AnnotationDbi")
library("org.Dm.eg.db")
```

BUSCO Analysis

```{r, echo=FALSE}
setwd("../raw_count_data/")

busco<-read.csv("busco_results_v2.csv")
busco.p <-read.csv("busco_percentages.csv")
bus.df<-read.csv("busco_percentages_df.csv")
bus.df$assemblies = factor(bus.df$assemblies,levels(bus.df$assemblies)[c(1,4,2,3)])


my_family <- "sans"
my_size_ratio <- 1
my_colors <- c("blue","forestgreen","purple")
my_bar_height <- 0.75
my_title <- "BUSCO Assessment Results"
my_values <- busco$full


```


You can see from this that the "Full" transcriptome retains a lot more BUSCOs (i.e. conserved genes) than our other three assemblies, so I've done all the analyses on this one. The first set of graphs and Venn Diagrams is what we've seen before, but I omitted the MA plots for Large vs Small females. I'm not sure that's even a very informative set of graphs, to be honest. 

```{r, echo=FALSE}
ggplot() +
  geom_bar(aes(y = my_percentage, x = assemblies, fill = category), data = bus.df, stat="identity", width=my_bar_height) +
  coord_flip() +
  theme_gray(base_size = 8) + 
  scale_y_continuous(labels = c("0","20","40","60","80","100"), breaks = c(0,20,40,60,80,100)) +
  scale_fill_manual(values = my_colors,labels =c("Complete (C)","Fragmented (F)"," Missing (M)")) +
  ggtitle(my_title) + 
  xlab("") + 
  ylab("%BUSCOs") + 
  theme(plot.title = element_text(family=my_family, colour = "black", size = rel(2.2)*my_size_ratio, face = "bold")) + 
  theme(legend.position="top",legend.title = element_blank()) + 
  theme(legend.text = element_text(family=my_family, size = rel(1.2)*my_size_ratio))
```


# DESeq Analysis

```{r, echo=FALSE, results="hide"}
setwd("../raw_count_data/")
experimental_design <- read.csv("express_design_matrix.csv",row.names = 1)


readcounts_full <- read.csv("full_index_reads_rounded_annot.csv", row.names = 1)


all.dds <- DESeqDataSetFromMatrix(readcounts_full, experimental_design , 
                                  design = formula(~ tissue + size + tissue:size))
all.dds$group<- factor(paste0(all.dds$tissue, all.dds$size, all.dds$sex))

design(all.dds) <- ~ group
all.dds <-DESeq(all.dds, quiet=TRUE)

lgvssm.m.hd<-results(all.dds, contrast=c("group", "hdhornlgmale", "hdhornsmmale"), alpha = 0.05)
lgvssm.m.th<-results(all.dds, contrast=c("group", "thxhornlgmale", "thxhornsmmale"), alpha = 0.05)
lgvssm.m.win<-results(all.dds, contrast=c("group", "wingslgmale", "wingssmmale"), alpha = 0.05)
lgvssm.m.gen<-results(all.dds, contrast=c("group", "genitalialgmale", "genitaliasmmale"), alpha = 0.05)


mvf.hd<-results(all.dds, contrast=list(c("grouphdhornlgmale","grouphdhornsmmale"), c("grouphdhornsmfemale","grouphdhornlgfemale")), listValues=c(1/2,-1/2), alpha = 0.05)
mvf.th<-results(all.dds, contrast=list(c("groupthxhornlgmale","groupthxhornsmmale"), c("groupthxhornsmfemale","groupthxhornlgfemale")), listValues=c(1/2,-1/2), alpha = 0.05)
mvf.win<-results(all.dds, contrast=list(c("groupwingslgmale","groupwingssmmale"), c("groupwingssmfemale","groupwingslgfemale")), listValues=c(1/2,-1/2), alpha = 0.05)


size.male.hd.sig <- as.data.frame(subset(lgvssm.m.hd, padj < 0.05))
size.male.th.sig <- as.data.frame(subset(lgvssm.m.th, padj < 0.05))
size.male.win.sig <- as.data.frame(subset(lgvssm.m.win, padj < 0.05))
size.male.gen.sig <- as.data.frame(subset(lgvssm.m.gen, padj < 0.05))


mvf.hd.sig<-as.data.frame(subset(mvf.hd, padj < 0.05))
mvf.th.sig<-as.data.frame(subset(mvf.th, padj < 0.05))
mvf.win.sig<-as.data.frame(subset(mvf.win, padj < 0.05))



de.size<- unique(c(rownames(size.male.hd.sig), rownames(size.male.th.sig), rownames(size.male.win.sig), rownames(size.male.gen.sig)))

de.sex<- unique(c(rownames(mvf.hd.sig), rownames(mvf.th.sig), rownames(mvf.win.sig)))


```

Nutritionally Sensitive MA plots. I.e. DE in large vs small males. Theres 187 gens DE in horns, 41 DE genes in Thoracic horns, 13 in Wings, and 25 in genitals. Almost our predictions, but genitals have more DE than wings. 

```{r, echo=FALSE}
par(mfrow= c(2,2))
plot(y = lgvssm.m.hd$log2FoldChange, x =log2(lgvssm.m.hd$baseMean) , 
     ylab = "log2 fold change (H/L)", xlab = "mean expression males",
     main = " MAplot males high vs low in head horns", pch = 1, col ="grey", ylim=c(-5,5), cex=0.75)
with(subset(lgvssm.m.hd, padj<.05 ), 
     points(y=log2FoldChange, x=log2(baseMean),
            pch=17, cex = 1, col="blue"))
abline(a=0, b=0 , col="black", lty=3)
abline(a=1,-1, b=0, col="black")


plot(y = lgvssm.m.th$log2FoldChange, x =log2(lgvssm.m.th$baseMean) , 
     ylab = "log2 fold change (H/L)", xlab = "mean expression males",
     main = " MAplot males high vs low in thoracic horns", pch = 1, col ="grey", ylim=c(-5,5), cex=0.75)
with(subset(lgvssm.m.th, padj<.05 ), 
     points(y=log2FoldChange, x=log2(baseMean),
            pch=17, cex = 1, col="purple"))
abline(a=0, b=0 , col="black", lty=3)
abline(a=1,-1, b=0, col="black")

plot(y = lgvssm.m.win$log2FoldChange, x =log2(lgvssm.m.win$baseMean) , 
     ylab = "log2 fold change (H/L)", xlab = "mean expression males",
     main = " MAplot males high vs low in wings", pch = 1, col ="grey", ylim=c(-5,5), cex=0.75)
with(subset(lgvssm.m.win, padj<.05 ), 
     points(y=log2FoldChange, x=log2(baseMean),
            pch=17, cex = 1, col="forestgreen"))
abline(a=0, b=0 , col="black", lty=3)
abline(a=1,-1, b=0, col="black")

plot(y = lgvssm.m.gen$log2FoldChange, x =log2(lgvssm.m.gen$baseMean) , 
     ylab = "log2 fold change (H/L)", xlab = "mean expression males",
     main = " MAplot males high vs low in genitals", pch =1, col ="grey", ylim=c(-5,5), cex=0.75)
with(subset(lgvssm.m.gen, padj<.05 ), 
     points(y=log2FoldChange, x=log2(baseMean),
            pch=17, cex = 1, col="red"))
abline(a=0, b=0 , col="black", lty=3)
abline(a=1,-1, b=0, col="black")
dev.off()
```

Nutritionally Sensitive Venn diagram. The genes DE in each tissue that overlap. As the initial draft says, these are pretty consistent. Horns share with horns but there isn't a LOT of overlap. 

```{r, echo=FALSE}
flog.threshold(ERROR)
male.nutrition<- (list(rownames(size.male.hd.sig), rownames(size.male.th.sig), rownames(size.male.win.sig), rownames(size.male.gen.sig)))
names(male.nutrition)<- c("Head", "Thorax", "Wings", 'Genitals')

vp.nutrition <- venn.diagram(male.nutrition, 
                             fill = c("blue", "purple", "green", "red"), cex=1.25, filename = NULL, main="Male Nutritionally Responsive Expression");
grid.draw(vp.nutrition)
```



Sexually Dimorphic Gene expression. We have 4031 seqs DE by sex in horns, 525 seqs DE by sex in Th horns, and 157 seqs DE by sex in wings, confirming our predictions. 


```{r, echo=FALSE}
par(mfcol= c(1,3))

plot(y = mvf.hd$log2FoldChange, x =log2(mvf.hd$baseMean) , 
     ylab = "log2 fold change (M/F)", xlab = "mean expression",
     main = " MAplot males Male vs Female head horns", pch = 1, cex=.75, col ="grey")
with(subset(mvf.hd, padj<.05 ), 
     points(y=log2FoldChange, x=log2(baseMean),
            pch=17, cex = 1, col="blue"))
abline(a=0, b=0 , col="black", lty=3)
abline(a=1,-1, b=0, col="black")

plot(y = mvf.th$log2FoldChange, x =log2(mvf.th$baseMean) , 
     ylab = "log2 fold change (M/F)", xlab = "mean expression",
     main = " MAplot males Male vs Female thoracic horns", pch = 1, col ="grey")
with(subset(mvf.th, padj<.05 ), 
     points(y=log2FoldChange, x=log2(baseMean),
            pch=17, cex = 1, col="purple"))
abline(a=0, b=0 , col="black", lty=3)
abline(a=1,-1, b=0, col="black")

plot(y = mvf.win$log2FoldChange, x =log2(mvf.win$baseMean) , 
     ylab = "log2 fold change (M/F)", xlab = "mean expression",
     main = " MAplot males Male vs Female wings", pch = 20, col ="grey")
with(subset(mvf.win, padj<.05 ), 
     points(y=log2FoldChange, x=log2(baseMean),
            pch=17, cex = 1, col="forestgreen"))
abline(a=0, b=0 , col="black", lty=3)
abline(a=1,-1, b=0, col="black")
dev.off()
```

Sexually Dimorphic Venns. There isn't a LOT of overlap in DE genes based on sexual dimorphism. Head horns and Thoracic horns share the greatest number of DE genes, but there's a core of 111 genes DE in all three tissues. 

```{r, echo=FALSE}
flog.threshold(ERROR)
sex.dimorph<- (list(rownames(mvf.hd.sig), rownames(mvf.th.sig), rownames(mvf.win.sig)))
names(sex.dimorph)<- c("Head", "Thorax", "Wings")

vp.sex <- venn.diagram(sex.dimorph, 
                             fill = c("blue", "purple", "green"), cex=1.25, filename = NULL, main="Sexually Dimorphic Gene Expression");
grid.draw(vp.sex)
```


Overlap between NR geneset and SD Geneset. I haven't had time to clean this up, but if we take ALL the genes differentially expressed as a result of sexual dimorphism and ALL the genes differentially expressed as a result of nutritional sensitivity, about half of the nutrition genes are also DE in the SD genes (approx 100 genes)
```{r, echo=FALSE}
flog.threshold(ERROR)
sex.nutrition<- (list(de.size, de.sex))
names(sex.nutrition)<- c("Nutritionally Sensitive","Sexually Dimorphic")

vp.sex.nutrition <- venn.diagram(sex.nutrition, 
                             fill = c("blue", "purple"), cex=1.25, filename = NULL, main="Overlap in Differentially Expressed Genesets");
grid.draw(vp.sex.nutrition)
```


```{r, echo=FALSE, results="hide"}
#The function below calculate Euclidean Distances (the L2 norm) so we can use unit vectors in the estimation of the vector correlation.
#These functions follow the logic of Kuruvilla et al 2002, and were adapted from Pitchers et al 2013
PD <- function(x) { 
  sqrt(t(x)%*%x)}


#this function gives me the vector correlation and angle, and vector magnitude ratio, alpha, between two vectors
#alpha of 1 means that vector 2 is the same length as vector 1. lower than one means that it is smaller, greater than 1 means that vector 2 is larger
ang.vec.alph <- function(vec1, vec2) {
  vec1 <- vec1 - mean(vec1)
  vec2 <- vec2 - mean(vec2)
  vec.cor <- abs((t(vec1) %*% vec2)/(PD(vec1)*PD(vec2)))
  vec.angle <- acos(vec.cor)*(180/pi)
  vec.alpha <- PD(vec1)/PD(vec2) # ID > RZ The absolute value was unnecessary since these are both magnitudes.
  vec.ED <- PD(vec2-vec1) #Subtract vector one from vector two and then calculate PD for Euclidean Distance. 
  return(c(vector.cor=vec.cor, vec.angle=vec.angle, vec.alpha=vec.alpha, vector.ED=vec.ED))} 


#The function below will generate two vectors of the same length as the experimental vectors provided to it with the argument 'sig_vec'

RandomVectorSampler <- function(vec1, vec2, sig_vec) {
  sig_vector_length <- length(sig_vec)
  
  random_index <- sample(1:length(vec1), size = sig_vector_length)
  sub_vec1 <- na.omit(vec1[random_index])
  sub_vec2 <- na.omit(vec2[random_index])
  
  ang.vec.alph(sub_vec1, sub_vec2)
}

comment(RandomVectorSampler) <- c("vec1 = vector of complete genelist comparison 1",
                                  "vec2 = vector of complete genelist comparison 2",
                                  "sig_vec = vector of significant genes, to get length")
```




```{r, echo=FALSE, results="hide"}

##Computing correlation for the SD gene set between the vector of fold changes of those genes between lgvsSm male horns and male vs female head horns
sd.size.male.vec1 <-subset(lgvssm.m.hd$log2FoldChange, rownames(lgvssm.m.hd) %in% de.sex)
mvf.horn.vec1 <-subset(mvf.hd$log2FoldChange, rownames(mvf.hd) %in% de.sex)


SD.size.horn.res<-as.data.frame(rbind(
  ang.vec.alph(mvf.horn.vec1, sd.size.male.vec1)), 
  rownames="estimates")

random_vectors_SSD_size<- t(replicate(10000, 
                                     RandomVectorSampler(mvf.hd$log2FoldChange, lgvssm.m.hd$log2FoldChange, de.sex)))

SD.size.horn.res<-rbind(SD.size.horn.res, apply(random_vectors_SSD_size, MARGIN = 2, quantile, probs = c(0.025, 0.5, 0.95), na.rm=TRUE))


SD.size.horn.res<-t(SD.size.horn.res) %>% as.data.frame() %>% rownames_to_column("value") %>% dplyr::rename(estimate="1") %>% mutate(., comparison="SD.mvf.horn")

##Computing correlation for the NR gene set between the vector of fold changes of those genes between lgvsSm male horns and male vs female head horns
NR.sd.vec1 <-subset(mvf.hd$log2FoldChange, rownames(mvf.hd) %in% de.size)
lgvsm.horn.vec1 <-subset(lgvssm.m.hd$log2FoldChange, rownames(lgvssm.m.win) %in% de.size)

NR.mvf.horn.res<-as.data.frame(rbind(ang.vec.alph(lgvsm.horn.vec1, NR.sd.vec1)), rownames="estimates")

random_vectors_NR_sd<- t(replicate(10000, 
                                     RandomVectorSampler(mvf.hd$log2FoldChange, lgvssm.m.hd$log2FoldChange, de.size)))

NR.mvf.horn.res<-rbind(NR.mvf.horn.res, apply(random_vectors_NR_sd, MARGIN = 2, quantile, probs = c(0.025, 0.5, 0.95), na.rm=TRUE))  

NR.mvf.horn.res<-t(NR.mvf.horn.res) %>% as.data.frame() %>% rownames_to_column("value") %>% dplyr::rename(estimate="1") %>% mutate(., comparison="NR.mvf.horn")

##computing correlations for the NR gene set between Head horns and Wings in males
#use the NR vector above, lgvsm.horn.vec1
#generate new vector of NR genes in the lg vs sm wings
lgvsm.wing.vec1 <-subset(lgvssm.m.win$log2FoldChange, rownames(lgvssm.m.win) %in% de.size )

NR.horn.wing.res<-as.data.frame(rbind(ang.vec.alph(lgvsm.horn.vec1, lgvsm.wing.vec1)), rownames="estimates")

random_vectors_NR_wing<- t(replicate(10000, 
                                     RandomVectorSampler(lgvssm.m.win$log2FoldChange, lgvssm.m.hd$log2FoldChange, de.size)))

NR.horn.wing.res<-rbind(NR.horn.wing.res, apply(random_vectors_NR_wing, MARGIN = 2, quantile, probs = c(0.025, 0.5, 0.95), na.rm=TRUE))  


NR.horn.wing.res<-t(NR.horn.wing.res) %>% as.data.frame() %>% rownames_to_column("value") %>% dplyr::rename(estimate="1") %>% mutate(., comparison="NR.horn.wing")

##computing correlations for the NR gene set between Head horns and genitals in males
#use the NR vector above, lgvsm.horn.vec1
#generate new vector of NR genes in the lg vs sm genitals
lgvsm.gen.vec1 <-subset(lgvssm.m.gen$log2FoldChange, rownames(lgvssm.m.gen) %in% de.size)

NR.horn.genital.res<-as.data.frame(rbind(ang.vec.alph(lgvsm.horn.vec1, lgvsm.gen.vec1)), rownames="estimates")

random_vectors_NR_gen<- t(replicate(10000, 
                                     RandomVectorSampler(lgvssm.m.gen$log2FoldChange, lgvssm.m.hd$log2FoldChange, de.size)))

NR.horn.genital.res<-rbind(NR.horn.genital.res, apply(random_vectors_NR_gen, MARGIN = 2, quantile, probs = c(0.025, 0.5, 0.95), na.rm=TRUE))  

NR.horn.genital.res<-t(NR.horn.genital.res) %>% as.data.frame() %>% rownames_to_column("value") %>% dplyr::rename(estimate="1") %>% mutate(., comparison="NR.horn.genital")


#computing correlations of the SD gene set between lg and sm hh/th horn
#use vector sd.size.male.vec1, and make new vector for th horn

sd.size.male.th.vec1<-subset(lgvssm.m.th$log2FoldChange, rownames(lgvssm.m.th) %in% de.sex)

sd.horn.th.res<-as.data.frame(rbind(ang.vec.alph(sd.size.male.vec1, sd.size.male.th.vec1)), rownames="estimates")

random_vectors_SD_hh_th<- t(replicate(10000, 
                                     RandomVectorSampler(lgvssm.m.th$log2FoldChange, lgvssm.m.hd$log2FoldChange, de.sex)))

sd.horn.th.res<-rbind(sd.horn.th.res, apply(random_vectors_SD_hh_th, MARGIN = 2, quantile, probs = c(0.025, 0.5, 0.95), na.rm=TRUE))  

sd.horn.th.res<-t(sd.horn.th.res) %>% as.data.frame() %>% rownames_to_column("value") %>% dplyr::rename(estimate="1") %>% mutate(., comparison="sd.horn.th")


#Finally compute correlation of NR gene set between HH/Th in males
##Use the vector lgvsm.horn.vec1, and make a similar one for TH
lgvsm.th.vec1 <-subset(lgvssm.m.th$log2FoldChange, rownames(lgvssm.m.th) %in% de.size )

nr.horn.th.res<-as.data.frame(rbind(ang.vec.alph(lgvsm.horn.vec1, lgvsm.th.vec1 )), rownames="estimates")

random_vectors_NR_hh_th<- t(replicate(10000, 
                                     RandomVectorSampler(lgvssm.m.th$log2FoldChange, lgvssm.m.hd$log2FoldChange, de.size)))

nr.horn.th.res<-rbind(nr.horn.th.res, apply(random_vectors_NR_hh_th, MARGIN = 2, quantile, probs = c(0.025, 0.5, 0.95), na.rm=TRUE))  

nr.horn.th.res<-t(nr.horn.th.res) %>% as.data.frame() %>% rownames_to_column("value") %>% dplyr::rename(estimate="1") %>% mutate(., comparison="nr.horn.th")


```



```{r, echo=FALSE, results="hide"}
full.results<-rbind(SD.size.horn.res,NR.mvf.horn.res,NR.horn.wing.res, NR.horn.genital.res, sd.horn.th.res, nr.horn.th.res)
full.results<-full.results %>% dplyr::rename(y.min="2.5%") %>% dplyr::rename(y.max="95%") %>% dplyr::rename(mid="50%")

full.results<-full.results %>% mutate(geneset = ifelse(grepl("SD", comparison, ignore.case=TRUE), "SD", "NR")) %>% mutate(group= ifelse(grepl("mvf", comparison), "1", ifelse(grepl("th", comparison), "3", "2")))

full.results[sapply(full.results, is.character)] <- lapply(full.results[sapply(full.results, is.character)], 
                                       as.factor)


```


```{r, echo=FALSE, results="hide"}
#test vec cor and alpha (respectively)
full.results$pvalue<-c(
  ((sum(random_vectors_SSD_size[,1] > SD.size.horn.res[1,2])+1)/10001),
  ((sum(random_vectors_SSD_size[,2] > SD.size.horn.res[2,2])+1)/10001),
((sum(random_vectors_SSD_size[,3] > SD.size.horn.res[3,2])+1)/10001),
((sum(random_vectors_SSD_size[,4] > SD.size.horn.res[4,2])+1)/10001),


((sum(random_vectors_NR_sd[,1] > NR.mvf.horn.res[1,2])+1)/10001),
((sum(random_vectors_NR_sd[,2] > NR.mvf.horn.res[2,2])+1)/10001),
((sum(random_vectors_NR_sd[,3] > NR.mvf.horn.res[3,2])+1)/10001),
((sum(random_vectors_NR_sd[,4] > NR.mvf.horn.res[4,2])+1)/10001),


((sum(random_vectors_NR_wing[,1] > NR.horn.wing.res[1,2])+1)/10001),
((sum(random_vectors_NR_wing[,2] > NR.horn.wing.res[2,2])+1)/10001),
((sum(random_vectors_NR_wing[,3] > NR.horn.wing.res[3,2])+1)/10001),
((sum(random_vectors_NR_wing[,4] > NR.horn.wing.res[4,2])+1)/10001),


((sum(random_vectors_NR_gen[,1] > NR.horn.genital.res[1,2])+1)/10001),
((sum(random_vectors_NR_gen[,2] > NR.horn.genital.res[2,2])+1)/10001),
((sum(random_vectors_NR_gen[,3] > NR.horn.genital.res[3,2])+1)/10001),
((sum(random_vectors_NR_gen[,4] > NR.horn.genital.res[4,2])+1)/10001),


((sum(random_vectors_SD_hh_th[,1] > sd.horn.th.res[1,2])+1)/10001),
((sum(random_vectors_SD_hh_th[,2] > sd.horn.th.res[2,2])+1)/10001),
((sum(random_vectors_SD_hh_th[,3] > sd.horn.th.res[3,2])+1)/10001),
((sum(random_vectors_SD_hh_th[,4] > sd.horn.th.res[4,2])+1)/10001),

((sum(random_vectors_NR_hh_th[,1] > nr.horn.th.res[1,2])+1)/10001),
((sum(random_vectors_NR_hh_th[,2] > nr.horn.th.res[2,2])+1)/10001),
((sum(random_vectors_NR_hh_th[,3] > nr.horn.th.res[3,2])+1)/10001),
((sum(random_vectors_NR_hh_th[,4] >nr.horn.th.res[4,2])+1)/10001))
correlation.alphas<-subset(full.results, value=="vector.cor" | value=="vec.alpha")

```

This is simply a table of the Correlation (r) and Alpha values, with p-values. 
```{r}
correlation.alphas
```


Plot the Results. Same as we did last time. Results match what we found with the reduced transcriptomes in the initial draft.  The x-axis in the correlation plot is the same as the Alpha plot below it.

```{r, echo=FALSE}

A<-ggplot(subset(full.results, value=="vector.cor"), aes(x = comparison, y = estimate ))+
  geom_crossbar(aes(ymin=y.min, ymax=y.max, fill=geneset))+
  labs(fill='Gene Vector')+
  ylab("Vector Correlation (r)")+
  coord_cartesian(ylim = c(0, 1))+ 
  theme(axis.text.x = element_blank(), strip.text.x=element_blank())+
  facet_grid(.~ group, scales="free_x")+
  guides(alpha=FALSE, xlab=FALSE)+
  xlab("")


B<-ggplot(subset(full.results, value=="vec.alpha"), aes(x = comparison, y = estimate ))+
  geom_crossbar(aes(ymin=y.min, ymax=y.max, fill=geneset))+
  labs(fill='Gene Vector')+
  ylab("Alpha")+
  coord_cartesian(ylim = c(0, 2))+ 
  theme(axis.text.x = element_blank(), strip.text.x = element_blank())+
  facet_grid(.~ group, scales="free_x")+
  guides(alpha=FALSE, xlab=FALSE)+
  xlab("")

B.labels<-ggplot(subset(full.results, value=="vec.alpha"), aes(x = comparison, y = estimate ))+
  geom_crossbar(aes(ymin=y.min, ymax=y.max, fill=geneset))+
  labs(fill='Gene Vector')+
  ylab("Alpha")+
  coord_cartesian(ylim = c(0, 2))+ 
  theme(axis.text.x = element_text(angle = 90, hjust = 1), strip.text.x = element_blank())+
  facet_grid(.~ group, scales="free_x")

par(mfrow=c(2,1))
A
B.labels

```


In the plot below, I computed the mean and SD for each gene in each group (i.e. the mean and SD for all the genes from large, male horns). I used either the differentially expressed genes by size, or all the genes in total. This is supposed to try and respond to our reviewer comment about "whether or not we simply captured the genes that are the most variable". As you can see, the SD of the DE genes is a bit more spread out than all of the genes, but the range on the SD is still pretty small. I'm still not sure I made this graph correctly, Ian please check this!

```{r}
rld <- rlog(all.dds)
rld.assay <-as.data.frame(assay(rld))


##lg male hd horn sd

degenes.sd<-as.data.frame(rld.assay %>%
  rownames_to_column('gene')%>%
  filter(gene %in% de.size)%>%
  column_to_rownames('gene')%>%
  summarise_all(., sd)%>%
  gather(sample, sd))

degenes.mean<-rld.assay %>%
  rownames_to_column('gene')%>%
  filter(gene %in% de.size)%>%
  column_to_rownames('gene')%>%
  summarise_all(., mean)%>%
  gather(sample, mean)%>%
  dplyr::select(., 2)
degenes.sd.mean<-cbind(degenes.sd, degenes.mean)
degenes.sd.mean<-degenes.sd.mean%>%
  separate(., sample, sep="_", c("Number", "size", "sex", "tissue"), remove=TRUE)%>%
  dplyr::select(., -Number)



allgenes.mean<-rld.assay%>%
  summarise_all(., mean)%>%
  gather(sample, mean)%>%
  dplyr::select(., 2)

allgenes.sd<-rld.assay%>%
  summarise_all(., sd)%>%
  gather(sample, sd)

allgenes.sd.mean<- cbind(allgenes.sd, allgenes.mean)
allgenes.sd.mean<-allgenes.sd.mean%>%
  separate(., sample, sep="_", c("Number", "size", "sex", "tissue"), remove=TRUE)%>%
  dplyr::select(., -Number)
  

summed.degenes<-group_by(degenes.sd.mean, size, sex, tissue) %>% summarise_all(funs(cum.mean=mean)) %>% dplyr::rename(grpmean=mean_cum.mean, grpsd=sd_cum.mean)
summed.allgenes<-group_by(allgenes.sd.mean, size, sex, tissue) %>% summarise_all(funs(cum.mean=mean)) %>% dplyr::rename(grpmean=mean_cum.mean, grpsd=sd_cum.mean)


ggplot(summed.degenes, aes(x=grpsd, y=grpmean))+
  geom_point(aes(color="de"))+
  geom_point(data=summed.allgenes, aes(x=grpsd, y=grpmean, color="all"))+
  ggtitle("Grouped Mean Expression vs Grouped SD for genes DE by size, by sample")+
  scale_color_discrete("Subset")



```

#plots for Reaction Norms


```{r,echo=FALSE}

sizecounts <-t(abs(log2((counts(all.dds[de.size, ], 
                         normalized=TRUE, replaced=FALSE)+0.5)))) %>%
  merge(colData(all.dds), ., by="row.names") %>%
  gather(gene, expression, (ncol(.)-length(de.size)+1):ncol(.))
levels(sizecounts$tissue) <- c("hdhorn", "thxhorn", "wings", "genitalia")
sizecounts$gene<-as.factor(sizecounts$gene)
##

sexcounts <-t(abs(log2((counts(all.dds[de.sex, ], 
                            normalized=TRUE, replaced=FALSE)+0.5)))) %>%
  merge(colData(all.dds), ., by="row.names") %>%
  gather(gene, expression, (ncol(.)-length(de.sex)+1):ncol(.))%>%
  filter(tissue != "genitalia")
levels(sexcounts$tissue)<-droplevels(sexcounts$tissue)
levels(sexcounts$tissue) <- c("hdhorn", "thxhorn", "wings")
sexcounts$gene<-as.factor(sexcounts$gene)

```

I like this graph, it seems to show nearly consistent expression of nutritionally sensitive genes across tissue types.

So what I did is take the genes DE by nutrition, and calculate the mean expression value for each gene across each tissue type. I used the absolute value of the log2 of the normalized counts, plus a 1/2 pseudocount. Then I took the mean of that value for each tissue for genes differentially expressed by size, and graphed how each gene changed across each of the four tissue types.  
```{r, echo=FALSE}


ggplot((group_by(sizecounts, gene, tissue) %>% dplyr::summarise(mean=mean(expression))), aes(x=tissue, y=mean, color=tissue))+
  geom_point()+geom_line(aes(group=interaction(gene)))+
  guides(color=FALSE)+
  scale_x_discrete(labels=c("Head Horn", "Thorax Horn", "Wings", "Genitals"))+
  labs(x="Tissue", 
       y="Expression (log normalized counts)", 
       title="Gene Reaction Norms for Nutritionally Sensitive Genes")


```

This graph is....not my favorite. I did the same as above, but I don't know how to make it look better or be more informative. It's just impossible to parse due to how many genes there are. Ideas for how to subset this would be nice!

```{r, echo=FALSE}
ggplot((group_by(sexcounts, gene, tissue) %>% dplyr::summarise(mean=mean(expression))), aes(x=tissue, y=mean, color=tissue))+
  geom_point()+geom_line(aes(group=interaction(gene)))+
  guides(color=FALSE)+
  scale_x_discrete(labels=c("Head Horn", "Thorax Horn", "Wings"))+
  labs(x="Tissue", 
       y="Expression (log normalized counts)", 
       title="Gene Reaction Norms for Sexually Dimorphic Genes")
```


 
```{r, results="hide", echo=FALSE}
##This code subsets the transcriptome to only get the sequences I want, and writes them into a text file so I can BLAST them later.
setwd("C:/Users/rober/Dropbox/BeetleRNASeq/TranscriptomeAssembliesForDoug")
full<-read.fasta("Trinity_m160_m172_f218_f135.fasta", seqtype = "AA", as.string = TRUE, set.attributes = FALSE)

horn_nr_seqs<-full[c(which(names(full) %in% rownames(size.male.hd.sig)))]
thhorn_nr_seqs<-full[c(which(names(full) %in% rownames(size.male.th.sig)))]
wing_nr_seqs<-full[c(which(names(full) %in% rownames(size.male.win.sig)))]
gen_nr_seqs<-full[c(which(names(full) %in% rownames(size.male.gen.sig)))]
##all unique DE NR genes (no duplicates)
nr_de_seqs<- full[c(which(names(full) %in% de.size))]

horn_sd_seqs<-full[c(which(names(full)%in% rownames(mvf.hd.sig)))]
thhorn_sd_seqs<-full[c(which(names(full)%in% rownames(mvf.th.sig)))]
wing_sd_seqs<-full[c(which(names(full)%in% rownames(mvf.win.sig)))]
#all unique DE SD genes (no duplicates)
sd_de_seqs<- full[c(which(names(full) %in% de.sex))]

setwd("C:/Users/rober/Dropbox/BeetleRNASeq/manuscript_files/manuscript_drafts/resubmission_files/tables/de_fasta_files_full")

write.fasta(sequences = horn_nr_seqs, names=names(horn_nr_seqs), file.out="size_DE_genes_horns.fasta")
write.fasta(sequences = thhorn_nr_seqs, names=names(thhorn_nr_seqs), file.out="size_DE_genes_thhorns.fasta")
write.fasta(sequences = wing_nr_seqs, names=names(wing_nr_seqs), file.out="size_DE_genes_wing.fasta")
write.fasta(sequences = gen_nr_seqs, names=names(gen_nr_seqs), file.out="size_DE_genes_gen.fasta")
write.fasta(sequences = nr_de_seqs, names=names(nr_de_seqs), file.out="nutritionally_responsive_DE_genes.fasta")

write.fasta(sequences = horn_sd_seqs, names=names(horn_sd_seqs), file.out="sd_DE_genes_horns.fasta")
write.fasta(sequences = thhorn_sd_seqs, names=names(thhorn_sd_seqs), file.out="sd_DE_genes_thhorns.fasta")
write.fasta(sequences = wing_sd_seqs, names=names(wing_sd_seqs), file.out="sd_DE_genes_wing.fasta")
write.fasta(sequences = sd_de_seqs, names=names(sd_de_seqs), file.out="sexually_dimorphic_genes.fasta")


#ignore this pseudocode to turn the output into a workable list and combine with results
#df <- ldply (gen_nr_seqs, data.frame, .id="gene")
#names(df)[2] <- "sequence"
#df$sequence<- as.character(df$sequence)
#df2<-rownames_to_column(size.male.gen.sig, 'gene') %>% inner_join( df, by = "gene")
```

The graphs below are an attempt to also address the "interaction" effects our reviewers were curious about. I'm not sure whether these address their concerns, and am going to lean on Ian. 

```{r, echo=FALSE, results="hide"}

readcounts_horn <- readcounts_full[, grep(".*hdhorn", colnames(readcounts_full),)]

expdesign_horn <- experimental_design[grep("hdhorn", experimental_design$tissue),]
colnames(readcounts_horn) == row.names(expdesign_horn)


horn.dds <- DESeqDataSetFromMatrix(readcounts_horn, expdesign_horn , 
                                  design = formula(~ sex + size + size:sex))

horn.dds <-DESeq(horn.dds, quiet=TRUE)



##thoracic horn
readcounts_th <- readcounts_full[, grep(".*thxhorn", colnames(readcounts_full),)]

expdesign_th <- experimental_design[grep("thxhorn", experimental_design$tissue),]
colnames(readcounts_th) == row.names(expdesign_th)

th.dds <- DESeqDataSetFromMatrix(readcounts_th, expdesign_th , 
                                   design = formula(~ sex + size + size:sex))

th.dds <-DESeq(th.dds, quiet=TRUE)




##wings
readcounts_win <- readcounts_full[, grep(".*wings", colnames(readcounts_full),)]

expdesign_win <- experimental_design[grep("wings", experimental_design$tissue),]
colnames(readcounts_win) == row.names(expdesign_win)

win.dds <- DESeqDataSetFromMatrix(readcounts_win, expdesign_win , 
                                 design = formula(~ sex + size + size:sex))

win.dds <-DESeq(win.dds, quiet=TRUE)




```
So, I subsetted the full data by tissue (head horn, thoracic horns, or wings). I used the model design = formula(~ sex + size + size:sex). Finally, I called results() on the full .dds object, and if I recall, that returns just the final factor in the model, i.e. size:sex, i.e. the interaction term. What's weird is we get confusing results. Thoracic horns have the most DE, followed by wings, followed by head horns. 


```{r}
summary(results(horn.dds, alpha=0.05))
summary(results(th.dds, alpha=0.05))
summary(results(win.dds, alpha=0.05))
```

```{r, echo=FALSE}
horn.de<-subset(results(horn.dds, alpha=0.05), padj <0.05)
th.de<-subset(results(th.dds, alpha=0.05), padj <0.05)
win.de<-subset(results(win.dds, alpha=0.05), padj <0.05)


```
Here's a Venn diagram of the overlap in DE genes in the interactions 
```{r, echo=FALSE}
de_contigs<- (list(rownames(horn.de), rownames(th.de), rownames(win.de)))
names(de_contigs)<- c("Head Horn", "Th Horn", "Wings")
vp.contigs <- venn.diagram(de_contigs, 
                                  fill = c("blue", "purple", "green"), cex=1.25, filename = NULL, main="Overlap in Interaction DE Genes");
grid.draw(vp.contigs)
```

The code below is the reaction norm of the intraction terms

```{r}



hi.counts <-t(abs(log2((counts(horn.dds[rownames(horn.de), ], 
                         normalized=TRUE, replaced=FALSE)+0.5)))) %>%
  merge(colData(horn.dds), ., by="row.names") %>%
  gather(gene, expression, (ncol(.)-length(rownames(horn.de))+1):ncol(.)) %>%
  rename(samples=Row.names) %>%
  mutate(samples=as.character(samples))%>%
  mutate(gene=factor(gene))



ti.counts <- t(abs(log2((counts(th.dds[rownames(th.de), ],
                                normalized=TRUE, replaced=FALSE)+0.5)))) %>%
  merge(colData(th.dds), ., by="row.names") %>%
  gather(gene, expression, (ncol(.)-length(rownames(th.de))+1):ncol(.))%>%
  rename(samples=Row.names) %>%
  mutate(samples=as.character(samples))%>%
  mutate(gene=factor(gene))



wi.counts <- t(abs(log2((counts(win.dds[rownames(win.de), ],
                                normalized=TRUE, replaced=FALSE)+0.5)))) %>%
  merge(colData(win.dds), ., by="row.names") %>%
  gather(gene, expression, (ncol(.)-length(rownames(win.de))+1):ncol(.))%>%
  rename(samples=Row.names) %>%
  mutate(samples=as.character(samples))%>%
  mutate(gene=factor(gene))




interact.counts <- bind_rows(list(horn = hi.counts, thx = ti.counts, wing=wi.counts), .id = "compare")


##need to work on this below
ggplot((group_by(interact.counts, gene, sex, size, tissue) %>% dplyr::summarise(mean=mean(expression)) %>% unite("condition", size , sex, sep="")), aes(x=condition, y=mean, color=tissue))+
  geom_point()+geom_line(aes(group=interaction(gene)))+
  #scale_x_discrete(labels=c("Head Horn", "Thorax Horn", "Wings"))+
  labs(x="condition", 
       y="Expression (log normalized counts)")+
  facet_grid(~tissue)



ggplot((group_by(interact.counts, gene, sex, size, tissue) %>% dplyr::summarise(mean=mean(expression)) %>% unite("condition", size , sex, sep="") %>% group_by(tissue, condition)%>% dplyr::summarise(tissuemean=mean(mean)) %>% mutate(condition = factor(condition, levels=c("lgmale", "smmale", "lgfemale", "smfemale")))), aes(x=condition, y=tissuemean, color=tissue))+
  geom_point()+geom_line(aes(group=interaction(tissue)))+
  #scale_x_discrete(labels=c("Head Horn", "Thorax Horn", "Wings"))+
  labs(x="condition", 
       y="Mean Expression (log normalized counts)")+
  facet_grid(~tissue)




##
```




```{r, echo=FALSE, results="hide"}
all.dds <- DESeqDataSetFromMatrix(readcounts_full, experimental_design , 
                                  design = formula(~ tissue + size + tissue:size))
all.dds$group<- factor(paste0(all.dds$tissue, all.dds$size, all.dds$sex))

design(all.dds) <- ~ group
all.dds <-DESeq(all.dds, quiet=TRUE)
```

Below, I tried to address the idea Ian had about running large tissues vs all other tissues (i.e. large male horns vs large female horns, small female horns, and small male horns combined). From what I've read, when doing an analysis of this type you need to adjust the denominator to take the average value of everything specified to it, so I've done so with listValues. I'm not sure what to make of all the data in this report, however, or even if it was the right way to handle this. It SEEMS to give us the results I want! Hh > Th horns > Wings, but on the other hand, pulling the interaction terms specifically, above, gives a different picture (i.e. TH > Wings > Hh). 
```{r}
horn.vsall<-results(all.dds, 
contrast=list("grouphdhornlgmale", c("grouphdhornsmmale","grouphdhornsmfemale","grouphdhornlgfemale")), listValues=c(1,-1/3), alpha=0.05)

summary(horn.vsall)

thx.vsall<-results(all.dds, 
contrast=list("groupthxhornlgmale", c("groupthxhornsmmale","groupthxhornsmfemale","groupthxhornlgfemale")), listValues=c(1,-1/3), alpha=0.05)

summary(thx.vsall)

wings.vsall<-results(all.dds, 
contrast=list("groupwingslgmale", c("groupwingssmmale","groupwingssmfemale","groupwingslgfemale")), listValues=c(1,-1/3), alpha=0.05)

summary(wings.vsall)

```

```{r, echo=FALSE}
hornvsall.de<-subset(horn.vsall, padj <0.05)
thxvsall.de<-subset(thx.vsall, padj <0.05)
wingsvsall.de<-subset(wings.vsall, padj <0.05)

de_contigs_vsall<- (list(rownames(hornvsall.de), rownames(thxvsall.de), rownames(wingsvsall.de)))
names(de_contigs_vsall)<- c("Head Horn", "Th Horn", "Wings")
vp.contigs_vsall <- venn.diagram(de_contigs_vsall, 
                                  fill = c("blue", "purple", "green"), cex=1.25, filename = NULL, main="Overlap in Interaction DE Genes contrasting Lg vs All");
grid.draw(vp.contigs_vsall)

```


```{r, echo=FALSE}
#This code is the start of me trying to pull out and write tables of annotations with gene names and fold changes etc

##head horn nutrition
size.male.hd.sig$FBpp <- sapply(strsplit(rownames(size.male.hd.sig), "\\|"), `[`, 2)
size.male.hd.sig$FBgn =   mapIds(org.Dm.eg.db,
                                 keys=size.male.hd.sig$FBpp , 
                                 column="FLYBASE",
                                 keytype="FLYBASEPROT",
                                 multiVals="first")

size.male.hd.sig$symbol = mapIds(org.Dm.eg.db,
                                 keys=size.male.hd.sig$FBpp , 
                                 column="SYMBOL",
                                 keytype="FLYBASEPROT",
                                 multiVals="first")
size.male.hd.sig$name =   mapIds(org.Dm.eg.db,
                                 keys=size.male.hd.sig$FBpp, 
                                 column="GENENAME",
                                 keytype="FLYBASEPROT",
                                 multiVals="first")

##th horn nutrition
size.male.th.sig$FBpp <- sapply(strsplit(rownames(size.male.th.sig), "\\|"), `[`, 2)
size.male.th.sig$FBgn =   mapIds(org.Dm.eg.db,
                                 keys=size.male.th.sig$FBpp , 
                                 column="FLYBASE",
                                 keytype="FLYBASEPROT",
                                 multiVals="first")

size.male.th.sig$symbol = mapIds(org.Dm.eg.db,
                                 keys=size.male.th.sig$FBpp , 
                                 column="SYMBOL",
                                 keytype="FLYBASEPROT",
                                 multiVals="first")
size.male.th.sig$name =   mapIds(org.Dm.eg.db,
                                 keys=size.male.th.sig$FBpp, 
                                 column="GENENAME",
                                 keytype="FLYBASEPROT",
                                 multiVals="first")


#wing nutrition
size.male.win.sig$FBpp <- sapply(strsplit(rownames(size.male.win.sig), "\\|"), `[`, 2)
size.male.win.sig$FBgn =   mapIds(org.Dm.eg.db,
                                 keys=size.male.win.sig$FBpp , 
                                 column="FLYBASE",
                                 keytype="FLYBASEPROT",
                                 multiVals="first")

size.male.win.sig$symbol = mapIds(org.Dm.eg.db,
                                 keys=size.male.win.sig$FBpp , 
                                 column="SYMBOL",
                                 keytype="FLYBASEPROT",
                                 multiVals="first")
size.male.win.sig$name =   mapIds(org.Dm.eg.db,
                                 keys=size.male.win.sig$FBpp, 
                                 column="GENENAME",
                                 keytype="FLYBASEPROT",
                                 multiVals="first")


#genital nutrition
size.male.gen.sig$FBpp <- sapply(strsplit(rownames(size.male.gen.sig), "\\|"), `[`, 2)
size.male.gen.sig$FBgn =   mapIds(org.Dm.eg.db,
                                 keys=size.male.gen.sig$FBpp , 
                                 column="FLYBASE",
                                 keytype="FLYBASEPROT",
                                 multiVals="first")

size.male.gen.sig$symbol = mapIds(org.Dm.eg.db,
                                 keys=size.male.gen.sig$FBpp , 
                                 column="SYMBOL",
                                 keytype="FLYBASEPROT",
                                 multiVals="first")
size.male.gen.sig$name =   mapIds(org.Dm.eg.db,
                                 keys=size.male.gen.sig$FBpp, 
                                 column="GENENAME",
                                 keytype="FLYBASEPROT",
                                 multiVals="first")


##male vs female horns
mvf.hd.sig$FBpp <- sapply(strsplit(rownames(mvf.hd.sig), "\\|"), `[`, 2)
mvf.hd.sig$FBgn =   mapIds(org.Dm.eg.db,
                                 keys=mvf.hd.sig$FBpp , 
                                 column="FLYBASE",
                                 keytype="FLYBASEPROT",
                                 multiVals="first")

mvf.hd.sig$symbol = mapIds(org.Dm.eg.db,
                                 keys=mvf.hd.sig$FBpp , 
                                 column="SYMBOL",
                                 keytype="FLYBASEPROT",
                                 multiVals="first")
mvf.hd.sig$name =   mapIds(org.Dm.eg.db,
                                 keys=mvf.hd.sig$FBpp, 
                                 column="GENENAME",
                                 keytype="FLYBASEPROT",
                                 multiVals="first")

##mvf thoracic
mvf.th.sig$FBpp <- sapply(strsplit(rownames(mvf.th.sig), "\\|"), `[`, 2)
mvf.th.sig$FBgn =   mapIds(org.Dm.eg.db,
                                 keys=mvf.th.sig$FBpp , 
                                 column="FLYBASE",
                                 keytype="FLYBASEPROT",
                                 multiVals="first")

mvf.th.sig$symbol = mapIds(org.Dm.eg.db,
                                 keys=mvf.th.sig$FBpp , 
                                 column="SYMBOL",
                                 keytype="FLYBASEPROT",
                                 multiVals="first")
mvf.th.sig$name =   mapIds(org.Dm.eg.db,
                                 keys=mvf.th.sig$FBpp, 
                                 column="GENENAME",
                                 keytype="FLYBASEPROT",
                                 multiVals="first")

##mvf wings
mvf.win.sig$FBpp <- sapply(strsplit(rownames(mvf.win.sig), "\\|"), `[`, 2)
mvf.win.sig$FBgn =   mapIds(org.Dm.eg.db,
                                 keys=mvf.win.sig$FBpp , 
                                 column="FLYBASE",
                                 keytype="FLYBASEPROT",
                                 multiVals="first")

mvf.win.sig$symbol = mapIds(org.Dm.eg.db,
                                 keys=mvf.win.sig$FBpp , 
                                 column="SYMBOL",
                                 keytype="FLYBASEPROT",
                                 multiVals="first")
mvf.win.sig$name =   mapIds(org.Dm.eg.db,
                                 keys=mvf.win.sig$FBpp, 
                                 column="GENENAME",
                                 keytype="FLYBASEPROT",
                                 multiVals="first")
#I need to repeat the above code for all the data frames I'm interested in. I can't figure out how to functionalize it, so I'll have to do it manually

setwd("C:/Users/rober/Dropbox/BeetleRNASeq/manuscript_files/manuscript_drafts/resubmission_files/tables")

library(xlsx)
wb <- createWorkbook()
gene_lists<-list(size.male.hd.sig, size.male.th.sig, size.male.win.sig, size.male.gen.sig, mvf.hd.sig, mvf.th.sig, mvf.win.sig)
sheetnames <- c("size.male.hd.sig", "size.male.th.sig", "size.male.win.sig", "size.male.gen.sig", "mvf.hd.sig", "mvf.th.sig", "mvf.win.sig")
sheets <- lapply(sheetnames, createSheet, wb = wb)
void <- Map(addDataFrame, gene_lists, sheets)
saveWorkbook(wb, file = "de_genes_full_transcriptome.xlsx")


```

